src/Microsoft.ML.TimeSeries/IidAnomalyDetectionBase.cs

```diff
@@ -13,99 +13,147 @@
 namespace Microsoft.ML.TimeSeriesProcessing
 {
     /// <summary>
-    /// This transform computes the p-values and martingale scores for a supposedly i.i.d input sequence of floats. In other words, it assumes
+    /// The wrapper to transform that computes the p-values and martingale scores for a supposedly i.i.d input sequence of floats. In other words, it assumes
     /// the input sequence represents the raw anomaly score which might have been computed via another process.
     /// </summary>
-    public abstract class IidAnomalyDetectionBase : SequentialAnomalyDetectionTransformBase<Single, IidAnomalyDetectionBase.State>
+    public class IidAnomalyDetectionBaseWrapper : IStatefulTransformer, ICanSaveModel
     {
-        public IidAnomalyDetectionBase(ArgumentsBase args, string name, IHostEnvironment env)
-            : base(args, name, env)
-        {
-            InitialWindowSize = 0;
-            StateRef = new State();
-            StateRef.InitState(WindowSize, InitialWindowSize, this, Host);
-        }
+        public bool IsRowToRowMapper => Base.IsRowToRowMapper;
```

Comment:
Do we want to expose this function to users? Maybe internal-best-friend it?

Reply:
I think it comes from the IStatefulTransformer, so it has to be public.

---
In reply to: [254477753](https://github.com/dotnet/machinelearning/pull/2344#discussion_r254477753) [](ancestors = 254477753)

Reply:
It comes from ITransformer which is public. 

---
In reply to: [254485421](https://github.com/dotnet/machinelearning/pull/2344#discussion_r254485421) [](ancestors = 254485421,254477753)